src/s3_fetchers.rs

```diff
@@ -1,26 +1,31 @@
 use std::str::FromStr;
 
 use aws_sdk_s3::Client;
-use futures::stream::StreamExt;
 
 const ESTIMATED_SHARDS_COUNT: usize = 4;
 
 /// Queries the list of the objects in the bucket, grouped by "/" delimiter.
-/// Returns the list of blocks that can be fetched
-pub(crate) async fn list_blocks(
+/// Returns the list of block heights that can be fetched
+pub(crate) async fn list_block_heights(
     s3_client: &Client,
     s3_bucket_name: &str,
     start_from_block_height: crate::types::BlockHeight,
     number_of_blocks_requested: usize,
-) -> anyhow::Result<Vec<crate::types::BlockHeight>> {
+) -> Result<
+    Vec<crate::types::BlockHeight>,
+    crate::types::LakeError<aws_sdk_s3::error::ListObjectsV2Error>,
+> {
     tracing::debug!(
         target: crate::LAKE_FRAMEWORK,
-        "Fetching blocks from S3, after {}...",
+        "Fetching block heights from S3, after #{}...",
         start_from_block_height
     );
     let response = s3_client
         .list_objects_v2()
-        .max_keys((number_of_blocks_requested * (1 + ESTIMATED_SHARDS_COUNT)).try_into()?)
+        .max_keys(std::cmp::min(
+            (number_of_blocks_requested * (1 + ESTIMATED_SHARDS_COUNT)).try_into()?,
```

Comment:
`min(1, ESTIMATED_SHARDS_COUNT)`?

Reply:
The maximum number of keys to get from S3 is 1000, I wanted to ensure we don't pass anything bigger than that even if we theoretically can predict the behavior.

Reply:
@frol do you propose just to stop receiving it even via function argument?